### DISCUSSION TITLE
Frequent Bad Gateway on Laravel Projetcs

### DISCUSSION THREAD

**Discussion by henryavila** [leokhoa/laragon#403] [Q&A] [answered]
Hi, I have some big projects built on laravel, and I have erro 502 - Bad Gateway too frequent. It hapens many times per day.
I know that it's an nginx message to inform that the php process has some problem (crashed, probally). Any ideas how can I optimize my php config to avoid error 502?

**Comment by rescenic** [comment]
Reload apache, clear browser cache.

**Reply by henryavila** [reply]
thanks @rescenic, but I believe is much more than this.
First, I use nginx, not apache. And the idea os not to reload to solve teh 502 error, is to avoid the error.

Probally the fpm process is crashing, but I have enough resources, probaly som missconfiguration

**Reply by samuelpatro** [reply]
Same issue with nginx every 4th request crashes to 502

**Reply by andrewdwallo** [reply]
Yes, I usually have this issue as well, but only when the page is taking too long to load and/or when I am developing, and I have too many "ErrorExceptions" in a row (usually 2 to 3 in a row is all it takes)... I am not sure how to fix it though. With Apache, even if I happened to have too many "ErrorExceptions" in a row, a "Bad Gateway", or "404 Not Found" error/crash would never occur. But the main reason I switched to Nginx is because I started to notice more and more problems with Apache regarding Laravel. I do not think Apache is suitable for Laravel anymore now (or never was), and I believe it has something to do with the Apache module (`mod_php`). But as right now, Nginx (w/ FastCGI and Gzip) and PHP NTS seem like the best fit. I wish I could solve this one issue though.

**Comment by jeddenkah** [comment]
Same problem here, any solution found?

**Comment by Hassanbhb** [comment]
I use Apache, when I installed laragon it worked fine, after a few days I turned on pc and it's giving me 502 error.
I reloaded the server, cleared cache, nothing.
can access the projects using Nginx but I have to add :8080 at the end of the link

I had to re install laragon to get the server working again. but the same thing happened again eventually.

**Reply by andrewdwallo** [reply]
When I was using Laragon for Windows (before switching to Mac), I recall having a more challenging time developing Laravel applications with Apache compared to Nginx. With Apache, I frequently encountered issues such as occasional blank screens. Another issue, which is detailed along with a solution in one of my apps [here](https://github.com/andrewdwallo/erpsaas/discussions/1#discussioncomment-7537349), also suggests using Nginx if you are working with Laragon.

**Reply by Hassanbhb** [reply]
Looks promising, I will check it when I get back home.

update: looks like I am sticking with Nginx.

**Comment by edo2313** [comment]
I had the same problem using Apache, then tried Nginx and it was even worse.
The logs were full of `upstream prematurely closed connection while reading response header from upstream` so I started digging in the nginx config files.
I found that in the file in `etc/nginx/php_upstream.conf` I had 2 entries: port 9003 and 9004. I tried leaving only the 9003 one and nothing worked, then I tried **only with the 9004** and so far I had no errors.

I hope this helps someone!

**Reply by edo2313** [reply]
Nevermind, the file is reset when you restart laragon.

**Reply by Norlandz** [reply, marked as answer]
Nice. I find the other processes are taking that port. I close those, and reload nginx in laragon. Then it works.

**Reply by leokhoa** [reply]
I have made a hotfix for that when fixing this issue: https://github.com/leokhoa/laragon/discussions/1440#discussioncomment-15711836
Please check the link for more information.
